services/121-service/src/registration/repositories/registration-view-scoped.repository.ts

```diff
@@ -15,4 +35,198 @@ export class RegistrationViewScopedRepository extends RegistrationScopedBaseRepo
+  ): ScopedQueryBuilder<RegistrationViewEntity> {
+    const relationInfo = attributeRelations.find((r) => r.name === sortByKey);
+    if (!relationInfo) {
+      return queryBuilder;
+    }
+    queryBuilder.leftJoin('registration.data', 'rd');
+    queryBuilder.andWhere(
+      new Brackets((qb) => {
+        this.whereRegistrationDataIsOneOfIds(relationInfo, qb, 'rd');
+      }),
+    );
+    queryBuilder.orderBy('rd.value', sortByValue);
+    queryBuilder.addSelect('rd.value');
+    // This is somehow needed (without alias!) to make the orderBy work
+    // These values are not returned because they are not mapped later on
+    return queryBuilder;
+  }
+
+  public filterRegistrationAttributeDataQb({
+    queryBuilder,
+    attributeRelations,
+    parsedFilter,
+  }: {
+    queryBuilder: ScopedQueryBuilder<RegistrationViewEntity>;
+    attributeRelations: RegistrationDataInfo[];
+    parsedFilter: ColumnsFilters;
+  }): ScopedQueryBuilder<RegistrationViewEntity> {
+    for (const [filterKey, filters] of Object.entries(parsedFilter)) {
+      const relationInfo = attributeRelations.find((r) => r.name === filterKey);
+      if (relationInfo) {
+        for (const filter of filters) {
+          queryBuilder = this.applySingleAttributeFilter({
+            queryBuilder,
+            filter,
+            relationInfo,
+          });
+        }
+      }
+    }
+    return queryBuilder;
+  }
+
+  private applySingleAttributeFilter({
+    queryBuilder,
+    filter,
+    relationInfo: relationInfo,
+  }: {
+    queryBuilder: ScopedQueryBuilder<RegistrationViewEntity>;
+    filter: Filter;
+    relationInfo: RegistrationDataInfo;
+  }): ScopedQueryBuilder<RegistrationViewEntity> {
+    const operatorTypes: FindOperatorType[] = [
+      'equal',
+      'in',
+      'ilike',
+      'isNull',
+      'moreThan',
+      'lessThan',
+      'between',
+    ];
+
+    const notOperatorType: FindOperatorType = 'not';
+
+    const findOperator = filter.findOperator;
+
+    const notFilter = findOperator.type === notOperatorType;
+
+    // This is needed to support nested find operators like $not:$ilike
+    const findOperatorType = findOperator.child?.type || findOperator.type;
+
+    if (operatorTypes.includes(findOperatorType)) {
+      const uniqueJoinId = Array.from({ length: 25 }, () =>
+        'abcdefghijklmnopqrstuvwxyz'.charAt(Math.floor(Math.random() * 26)),
+      ).join('');
+      queryBuilder.leftJoin(
+        'registration.data', // relation path
+        uniqueJoinId, // alias
+        `${uniqueJoinId}."programRegistrationAttributeId" = ${relationInfo.relation.programRegistrationAttributeId}`,
+      );
+      queryBuilder =
+        RegistrationFilterQueryHelpers.applyFilterConditionAttributes({
+          queryBuilder,
+          findOperatorType,
+          value: findOperator.value,
+          uniqueJoinId,
+          notFilter,
+        });
+    }
+    return queryBuilder;
+  }
+
+  private whereRegistrationDataIsOneOfIds(
```

Comment:
does this not make sense to move to the (repository) helper then also? 

To be honest, now that I'm looking at it again, I don't really get the split between what's here and what's in the helper. You said, the helper methods only create/modify queries, but actually all the methods in this repository also do that right? They all just return queryBuilders?